### PR TITLE
DIV-5807 "As a user, I expect to be able to attach files from OneDrive for Business"

### PR DESCRIPTION
**Pull request by vetostream**
Couple of changes here, made some adjustments on some provision calls and when creating instance where the element is onedrivebusiness. Also added an input where users can specify site address.

### CODE REVIEW

**Comment by vetostream** [comment]
@nmallare this is ready for review.

File: divvy/cloudfilebrowser.js [hunk 1]
```diff
@@ -212,6 +212,16 @@ var provision = (function() {
...
             if(elementDetails != null && elementDetails != undefined) {
 
+                // if onedrivebusiness
+                if (element === 'onedrivebusiness') {
+                    var siteAddress = cbArgs.siteAddress;
+                    if (siteAddress === null || siteAddress === "") {
+                        return
+                    } else {
+                        elementDetails.siteAddress = siteAddress;
+                    }
```
Comment (CollinEMac) [inline divvy/cloudfilebrowser.js:222]: We could use the double ! trick to make this section cleaner:

```
                    if (!!siteAddress) {
                        elementDetails.siteAddress = siteAddress;
                    } else {
                        return;
                    }
```

File: divvy/cloudfilebrowser.js [hunk 2]
```diff
@@ -612,6 +623,12 @@ var server = (function() {
                 'name': element
             };
 
+            if (element === 'onedrivebusiness') {
+                elementProvision.configuration["document.tagging"] = true;
+                elementProvision.configuration["filter.response.nulls"] = true;
+                elementProvision.configuration["onedrivebusiness.site.address"] = cbArgs.elementDetails.siteAddress;
```
Comment (CollinEMac) [inline divvy/cloudfilebrowser.js:631]: What are these parameters for and why do we only need them for `onedrivebusiness` elements? Maybe drop a comment explaining.

**Review by CollinEMac — DISMISSED** [review]

**Comment by CollinEMac** [comment]
Sorry this review took so long, @vetostream. I let if fall off my radar.

File: divvy/cloudfilebrowser.js (hunk 1 above)
Comment (vetostream) [reply divvy/cloudfilebrowser.js:222]: Good call.

**Pull request merged**